index.html

```diff
@@ -3,134 +3,477 @@
+      .domain([0, d3.max(yhist, function (d) { return d.length; })])
+      .range([0, height/barMax]);
+
+
+    var ybar = chart.svg.selectAll('.ybar')
+      .data(yhist)
+
+
+    //this is coming out way too big. dx for yhist is 2000, which is the correct value. The way I understand this syntax is that its converting that 2000 to a pixel value to fit along my earlier defined y scale.  This is the way it seemed to work for the x axis histogram, and also in calculating the position of each bar 
+
+    //Hacked this by dividing by 12. I'm still unsure why without the /12 it didnt come out to the right height.
+    var ybarHeight = chart.y(yhist[1].x-yhist[0].x)/(chart.y.ticks().length-1)+1.8;
+
+    var ybarSpace = ybarHeight/4;
+    //Positioning seems to be almost there, except its offset by one bar's length
+    //Update: Hacked this too by subtracting one bar height. 
+    
+    //ENTER
+    ybar
+      .enter().append('rect')
+      .attr('class', 'ybar')
+      .attr('height', ybarHeight-(1/2)*ybarHeight)
+      .attr('y', function (d) {return (chart.y(d.x)-ybarHeight) + ybarSpace;})
+      .on('mouseover', chart.barTooltip.show)
+      .on('mouseout', chart.barTooltip.hide);
+     
+
+    ybar
+      .transition().duration(1000).delay(150)
+      .attr('width', function (d) {return hx(d.length);})  //y scale
+      .attr('x', 0);
+
+
+   // HISTOGRAM X
+    var hist = d3.layout.histogram()
+      .bins(chart.x.ticks())
+      .value(function (d) {return d[chart.xvariable];})
+      (txData);
+
+    var hy = d3.scale.linear()
+      .domain([0, d3.max(hist, function (d) { return d.length; })])
+      .range([0, height/barMax]);
+
+    var bar = chart.svg.selectAll('.bar')
+      .data(hist)
+
+    var barWidth = chart.x(hist[0].dx)+1;
+    var xbarSpace = barWidth/4;
+   
+
+    bar
+      .enter().append('rect')
+      .attr('class', 'bar')
+      .attr('width', barWidth-(1/2)*barWidth)
+      .attr('x', function (d) {return (chart.x(d.x))+ xbarSpace;})
+      .on('mouseover', chart.barTooltip.show)
+      .on('mouseout', chart.barTooltip.hide);
+
+    bar
+      .transition().duration(1000).delay(150)
+      .attr('height', function (d) {return hy(d.length);})  //y scale
+      .attr('y', function (d) {return height - hy(d.length);});
+
+       
+    //No second chart histogram yet. Will work on converting to the constructor pattern.
+     
+
     // SCATTERPLOT
+    var points = chart.svg.selectAll('.point')
+      .data(txData);
+
+    points.enter().append('circle')
+      .attr('class', 'point')
+       .on('mouseover', chart.pointTooltip.show)
+       .on('mouseout', chart.pointTooltip.hide);
+
 
-    // HISTOGRAMS
+    points
+    .transition().duration(1000).delay(150)
+      .attr('cx', function (d) { return chart.x(d[chart.xvariable]); })
+      .attr('cy', function (d) { return chart.y(d[chart.yvariable]); })
+      .attr('r', function (d) { return rscale(d.Pop); });
+    points.exit().remove();
 
```

Comment:
Yes, definitely. I was going to ask how to do that. Will give it a few more
tries and then ask for help before/after class if I don't get it down.

Thanks,
Rich

On Thu, Jul 14, 2016 at 10:28 AM, Alex C Engler <notifications@github.com>
wrote:

> In index.html
> <https://github.com/JHU-DataViz-Summer16/week-4-replication-project/pull/7#discussion_r70815156>
> :
>
> >
> > -    // HISTOGRAMS
> > +    points
> > +    .transition().duration(1000).delay(150)
> > +      .attr('cx', function (d) { return chart.x(d[chart.xvariable]); })
> > +      .attr('cy', function (d) { return chart.y(d[chart.yvariable]); })
> > +      .attr('r', function (d) { return rscale(d.Pop); });
> > +    points.exit().remove();
> >
>
> Just to help illustrate what's happening, right now, some of your points
> are being updated and some are being created when you move between subsets.
> So, let's saying you move from a state with 10 points to a state with 20
> points.
>
> The first 10 points (first in the sense that they are the first 10 objects
> in the new array of data) are going to transition - make an animated move
> from the location under the old data to their new assignment. The other 10
> points are just going to appear where they are supposed to be. This makes
> the animation look a bit clunky - worth thinking about other options (like
> making all old points fade out, and all new points fade in).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/JHU-DataViz-Summer16/week-4-replication-project/pull/7/files/61abaeb620d1f7280d0db5ccf421750888f73957#r70815156>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AKBZdd_CoHBpO3gJE2mNZ75-vYn6ctITks5qVkdxgaJpZM4JBzHA>
> .
>